query/sources/impl/src/main/java/org/codice/ddf/admin/sources/wfs/persist/UpdateWfsConfiguration.java

```diff
@@ -0,0 +1,145 @@
+import org.codice.ddf.admin.configurator.Configurator;
+import org.codice.ddf.admin.configurator.ConfiguratorFactory;
+import org.codice.ddf.admin.configurator.OperationReport;
+import org.codice.ddf.admin.sources.fields.WfsVersion;
+import org.codice.ddf.admin.sources.fields.type.WfsSourceConfigurationField;
+import org.codice.ddf.admin.sources.utils.SourceUtilCommons;
+import org.codice.ddf.admin.sources.utils.SourceValidationUtils;
+import org.codice.ddf.internal.admin.configurator.actions.FeatureActions;
+import org.codice.ddf.internal.admin.configurator.actions.ManagedServiceActions;
+import org.codice.ddf.internal.admin.configurator.actions.ServiceActions;
+import org.codice.ddf.internal.admin.configurator.actions.ServiceReader;
+
+import com.google.common.collect.ImmutableList;
+
+public class UpdateWfsConfiguration extends BaseFunctionField<BooleanField> {
+
+    public static final String FIELD_NAME = "updateWfsSource";
+
+    private static final String DESCRIPTION =
+            "Updates a WFS source configuration specified by the pid. Returns true on success and false on failure.";
+
+    public static final BooleanField RETURN_TYPE = new BooleanField();
+
+    private WfsSourceConfigurationField config;
+
+    private SourceValidationUtils sourceValidationUtils;
+
+    private SourceUtilCommons sourceUtilCommons;
+
+    private final ConfiguratorFactory configuratorFactory;
+
+    private final ServiceActions serviceActions;
+
+    private final ManagedServiceActions managedServiceActions;
+
+    private final ServiceReader serviceReader;
+
+    private final FeatureActions featureActions;
+
+    public UpdateWfsConfiguration(ConfiguratorFactory configuratorFactory,
+            ServiceActions serviceActions, ManagedServiceActions managedServiceActions,
+            ServiceReader serviceReader, FeatureActions featureActions) {
+
+        super(FIELD_NAME, DESCRIPTION);
+        this.configuratorFactory = configuratorFactory;
+        this.serviceActions = serviceActions;
+        this.managedServiceActions = managedServiceActions;
+        this.serviceReader = serviceReader;
+        this.featureActions = featureActions;
+
+        config = new WfsSourceConfigurationField();
+        config.useDefaultRequired();
+        config.pidField()
+                .isRequired(true);
+        updateArgumentPaths();
+
+        sourceValidationUtils = new SourceValidationUtils(serviceReader,
+                managedServiceActions,
+                configuratorFactory,
+                serviceActions);
+        sourceUtilCommons = new SourceUtilCommons(managedServiceActions,
+                serviceActions,
+                serviceReader,
+                configuratorFactory);
+    }
+
+    @Override
+    public BooleanField performFunction() {
+        Configurator configurator = configuratorFactory.getConfigurator();
+        OperationReport report = null;
+        if (config.wfsVersion()
```

Comment:
If it is not one of the two versions, a failed persist will return. Also, if we have an invalid enum for the WfsVersion, we should catch it in validation (INVALID_ENUM) before performing the function.